src/MKL.jl

```diff
@@ -1,9 +1,48 @@
 module MKL
 
-using MKL_jll
+using Preferences
+using Libdl
 
-JULIA_VER_NEEDED = v"1.7.0-DEV.641"
-VERSION > JULIA_VER_NEEDED && using LinearAlgebra
+# Choose an MKL provider/path; taking an explicit preference as the first choice,
+# but if nothing is set as a preference, fall back to an environment variable,
+# and if that is not given, fall back to the default choice of `MKL_jll`.
+# Note: The environment variable is only read at (.ji) compile time and
+# not(!) every time before loading the package.
+const mkl_path = lowercase(something(
+    @load_preference("mkl_path", nothing),
+    get(ENV, "JULIA_MKL_PATH", nothing),
+    "mkl_jll",
+)::String)
+
+if lowercase(mkl_path) == "mkl_jll"
+    # Only load MKL_jll if we are suppoed to use it as the MKL source
+    # to avoid an unnecessary download of the (lazy) artifact.
+    import MKL_jll
+    const libmkl_rt = MKL_jll.libmkl_rt
+elseif lowercase(mkl_path) == "system"
+    # We expect the "system" MKL to already be loaded,
+    # or be on our linker search path.
+    libname = string("libmkl_rt", ".", Libdl.dlext)
+    const libmkl_rt = find_library(libname, [""])
+    libmkl_rt == "" && error("Couldn't find $libname. Try to set JULIA_MKL_PATH explicitly.")
+else
+    # mkl_path should be a valid path to libmkl_rt.
+    const libmkl_rt = mkl_path
+    isfile(libmkl_rt) || error("Couldn't find MKL library at location $libmkl_rt. JULIA_MKL_PATH needs to point directly to libmkl_rt.")
+end
+
+# Changing the MKL provider/path preference
+function set_mkl_path(path)
+    if lowercase(path) ∉ ("mkl_jll", "system")
+        isfile(path) || error("The provided argument $path doesn't seem to be a valid path to libmkl_rt.")
+    end
+    @set_preferences!("mkl_path" => path)
+    @info("New MKL provider/path set; please restart Julia to see this take effect", path)
+end
+
+is_lbt_available() = VERSION > v"1.7.0-DEV.641"
+
+is_lbt_available() && using LinearAlgebra
```

Comment:
It's safe to unconditionally `using LinearAlgebra` at the top of this file; it shouldn't have any performance impact.